modin/core/execution/utils.py

```diff
@@ -29,3 +31,62 @@ def set_env(**environ):
     finally:
         os.environ.clear()
         os.environ.update(old_environ)
+
+
+if "_MODIN_DOC_CHECKER_" in os.environ:
+
+    # The doc checker should get the non-processed functions
+    def remote_function(func, ignore_defaults=False):
+        return func
+
+
+# Check if the function already exists to avoid circular imports
+elif "remote_function" not in dir():
+    from modin.config import Engine
+
+    if Engine.get() == "Ray":
+        from modin.core.execution.ray.common import RayWrapper
+
+        _preprocess_func = RayWrapper.put
+    elif Engine.get() == "Unidist":
+        from modin.core.execution.unidist.common import UnidistWrapper
+
+        _preprocess_func = UnidistWrapper.put
+    elif Engine.get() == "Dask":
+        from modin.core.execution.dask.common import DaskWrapper
+
+        # The function cache is not supported for Dask
+        def remote_function(func, ignore_defaults=False):
+            return DaskWrapper.put(func)
+
+    else:
+
+        def remote_function(func, ignore_defaults=False):
+            return func
+
+    if "remote_function" not in dir():
+        _remote_function_cache = {}
+
+        def remote_function(func, ignore_defaults=False):  # noqa: F811
```

Comment:
I wonder how many warnings will we get using Ray engine with these changes?

Reply:
Why do you expect warnings here?

Reply:
Not here exactly but below we have a few conditions that can trigger a warning. I wonder how many of them will be?